### PR TITLE
Implement a very simple test case

### PR DESCRIPTION
**Pull request by TheOneRing**
Works on Windows and Linux.
Fails on Mac, probably access to the keychain is denied....

### CODE REVIEW

File: autotest/basic.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
+#include <QtTest>
+
+#include "qtkeychain/keychain.h"
+
+namespace
+{
+    QString serviceKey = QStringLiteral("QtKeychainTest");
```
Comment (frankosterfeld) [inline autotest/basic.cpp:7]: I wonder if this string should also contain a random component, so there can't be any clashes when running multiple builds/tests in parallel.
The risk would be leaving multiple "QtKeyChainTest-$randomHash" entries behind if the test repeatedly fails cleaning up, but I think that risk is small.

Yay for unit tests btw, finally time to set up github CI ;)

**Comment by frankosterfeld** [comment]
For being usable in CI context, it would probably be good to optionally disable the test for macOS (I wouldn't know how to get around manually confirming the popup)

**Comment by TheOneRing** [comment]
I didn't even get a popup it just failed for me.... that's why I added https://github.com/frankosterfeld/qtkeychain/pull/250/files#diff-130a09e7601618010ec6428acf245c2dddba455314877774bc3a3c0397f7303dR35
So every test run will scream at us that we should fix it...

File: autotest/basic.cpp (hunk 1 above)
Comment (TheOneRing) [reply autotest/basic.cpp:7]: I'll use a random key

File: autotest/basic.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+#include "qtkeychain/keychain.h"
+
+namespace
+{
+    QByteArray generateRandomString(qsizetype size)
+    {
+        std::vector<quint32> buffer(size, 0);
+        QRandomGenerator::global()->fillRange(buffer.data(), size);
+        return QByteArray(reinterpret_cast<char *>(buffer.data()), static_cast<int>(size * sizeof(quint32))).toBase64(QByteArray::Base64UrlEncoding).mid(0, size);
```
Comment (frankosterfeld) [inline autotest/basic.cpp:11]: Jeez, I somehow thought there was a "randomString" method somewhere, but maybe that was KDE :)

**Comment by frankosterfeld** [comment]
Thanks!

**Pull request merged**